examples/dnsdoctor/dnsdoctor.lua

```diff
@@ -0,0 +1,77 @@
+--
+-- SPDX-FileCopyrightText: (c) 2024 Mohammad Shehar Yaar Tausif <sheharyaar48@gmail.com>
+-- SPDX-License-Identifier: MIT OR GPL-2.0-only
+--
+
+-- DNS Doctoring : Rewrite DNS type A record to a private address for local clients
+
+local xt = require("xtable")
+local linux = require("linux")
+local string = require("string")
+local action = xt.action
+local family = xt.family
+
+local udp = 0x11
+local dns = 0x35
+
+local function nop() end
+
+local function get_domain(skb, off)
+	local name = ""
+	local i = 0
+
+	repeat
+		local d = skb:getbyte(off + i)
+		i = i + 1
+	until d == 0
```

Comment:
wouldn't it be better to `tostring(skb):find('\0')`?

Reply:
you mean because `off`? wouldn't be it just the case of `skb:getstring(off, len):find('\0')` or even `tostring(skb):sub(off + 1):find('\0')`? I miss the point of verifying it byte per byte instead of using a higher-level string API.. however, it's up to you.. we can merge it as is..

Reply:
Yeah, this works : `tostring(skb):sub(off + 1):find('\0')`

Reply:
Done in new push